### PR TITLE
Mahara autoscale caching

### PR DESCRIPTION
**Pull request by CatalystAUInf**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use resourceGroup().location for resource locations
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

- [ ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

**Comment by CatalystAUInf** [comment]
Ignore this pr.

**Comment by CatalystAUInf** [comment]
CI keeps failing with:

>   1) Template Running 1 Parallel Template Validation(s)... mahara-autoscale-caching:
      AssertionError: Template Validiation Failed. Try deploying your template with the commands:
azure group template validate --resource-group (your_group_name)  --template-file mahara-autoscale-caching/azuredeploy.json --parameters-file mahara-autoscale-caching/azuredeploy.parameters.json
azure group deployment create --resource-group (your_group_name)  --template-file mahara-autoscale-caching/azuredeploy.json --parameters-file mahara-autoscale-caching/azuredeploy.parameters.json 
Server Error:{
    "statusCode": 400,
    "body": {
        "error": "Error: stdout maxBuffer exceeded"
    },

Not sure what that means as it will build a mahara stack with either PG or MySQL.

**Comment by CatalystAUInf** [comment]
After running the suggested command from the cli logs, it is saying:

 azure group template validate --resource-group test01 --template-file mahara-autoscale-caching/azuredeploy.json --parameters-file mahara-autoscale-caching/azuredeploy.parameters.json
info:    Executing command group template validate
+ Initializing template configurations and parameters                          
+ Validating the template                                                      
error:   LocationNotAvailableForResourceType : The provided location 'australiaeast' is not available for resource type 'Microsoft.DBforMySQL/servers'. List of available regions for the resource type is 'brazilsouth,canadacentral,canadaeast,centralindia,eastasia,eastus2,eastus,japaneast,japanwest,northcentralus,northeurope,southcentralus,southeastasia,westeurope,westindia,westus'.
info:    group template validate command OK

How is it that mysql is not available in Australia?

File: mahara-autoscale-caching/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,757 @@
...
+            "vmssName": "[concat('vmss-',variables('resourceprefix'))]",
+            "vmssdStorageAccounttName": "[concat('vmss',uniqueString(resourceGroup().id))]",
+            "vnetGwDeploySwitch": "[parameters('vnetGwDeploySwitch')]",
+            "vnetName": "[concat('vnet-',variables('resourceprefix'))]",
+            "vpnType": "[parameters('vpnType')]",
+            "webServerSetupScriptFilename": "setup_webserver.sh",
+            "webServerType": "[parameters('webServerType')]",
+            "serviceObjective": "[parameters('serviceObjective')]",
+            "msDbSize": "[parameters('msDbSize')]",
+            "webServerType": "[parameters('webServerType')]"
```
Comment (bmoore-msft) [inline mahara-autoscale-caching/azuredeploy.json]: remove duplicate variable

File: mahara-autoscale-caching/nested/blobStorageAccount.json [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+                "supportsHttpsTrafficOnly": true
+            },
+            "type": "Microsoft.Storage/storageAccounts"
+        }
+    ],
+    "variables": {
+        "documentation1": "This sub-template creates a blob storage account.  It expects certain values in the 'common' datastructure.",
+        "documentation2": " blobStorageAccountName - name of storage account",
+        "documentation3": " blobStorageAccountType - type of storage account",
+        "scriptUri": "[concat(parameters('maharaCommon').ScriptLocation,parameters('maharaCommon').maharaInstallScriptFilename)]",
```
Comment (bmoore-msft) [inline mahara-autoscale-caching/nested/blobStorageAccount.json]: remove - not used

File: mahara-autoscale-caching/nested/network.json [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+        "documentation06": "    subnetSan        - name of subnet inside virtual network - will be assigned the .1.0/24 range",
+        "documentation07": "    subnetRedis      - name of subnet inside virtual network - will be assigned the .3.0/24 range",
+        "documentation08": "    subnetElastic    - name of subnet inside virtual network - will be assigned the .4.0/24 range",
+        "documentation09": "    gatewaySubnet    - name of subnet inside virtual network - will be assigned the .2.0/24 range",
+        "documentation10": "    lbPipName        - name of public IP",
+        "documentation11": "    lbName           - name of Moodl load balancer",
+        "extBeID": "[concat(variables('extLbID'),'/backendAddressPools/',parameters('maharaCommon').extBeName)]",
+        "extFeID": "[concat(variables('extLbID'),'/frontendIPConfigurations/',parameters('maharaCommon').extFeName)]",
+        "extLbID": "[resourceId('Microsoft.Network/loadBalancers',parameters('maharaCommon').lbName)]",
+        "extNatPoolID": "[concat(variables('extLbID'),'/inboundNatPools/',parameters('maharaCommon').extNatPool)]",
```
Comment (bmoore-msft) [inline mahara-autoscale-caching/nested/network.json]: remove unused variable

**Comment by CatalystAUInf** [comment]
HI, I've updated the URL and pushed, doesn't appear to be reflected here.

File: mahara-autoscale-caching/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,757 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "_artifactsLocation": {
+            "type": "string",
+            "metadata": {
+                "description": "The base URI where artifacts required by this template are located. When the template is deployed using the accompanying scripts, a private location in the subscription will be used and this value will be automatically generated."
+            },
+            "defaultValue": "https://raw.githubusercontent.com/CatalystAUInf/Mahara/master/"
```
[the remark this answers is not in the document]
Comment (CatalystAUInf) [reply mahara-autoscale-caching/azuredeploy.json]: Github content URL updated.

File: mahara-autoscale-caching/etc/checkBaseUrls.sh [hunk 5]
```diff
@@ -0,0 +1,58 @@
+# Ensure that the Base URL for templates, scripts and deploy to Azure buttons
+# is correctly set for the current git branch.
+
+# Correct values for locations
+CURRENT_BRANCH=$(git branch | sed -n -e 's/^\* \(.*\)/\1/p')
+echo "Current git branch is '$CURRENT_BRANCH'"
+
+BASE_TEMPLATE_URL=https://raw.githubusercontent.com/Azure/Moodle/$CURRENT_BRANCH/nested/
```
[the remark this answers is not in the document]
Comment (CatalystAUInf) [reply mahara-autoscale-caching/etc/checkBaseUrls.sh]: Updated.

**Comment by CatalystAUInf** [comment]
Closing this PR, superseded by https://github.com/Azure/azure-quickstart-templates/pull/4628 .

**Pull request closed without merging**